art/attacks/evasion/boundary.py

```diff
@@ -61,71 +62,57 @@ class BoundaryAttack(EvasionAttack):
         """
         Create a boundary attack instance.
 
         :param estimator: A trained classifier.
-        :type estimator: :class:`.ClassifierMixin`
         :param targeted: Should the attack target one specific class.
-        :type targeted: `bool`
         :param delta: Initial step size for the orthogonal step.
-        :type delta: `float`
         :param epsilon: Initial step size for the step towards the target.
-        :type epsilon: `float`
         :param step_adapt: Factor by which the step sizes are multiplied or divided, must be in the range (0, 1).
-        :type step_adapt: `float`
         :param max_iter: Maximum number of iterations.
-        :type max_iter: `int`
         :param num_trial: Maximum number of trials per iteration.
-        :type num_trial: `int`
         :param sample_size: Number of samples per trial.
-        :type sample_size: `int`
         :param init_size: Maximum number of trials for initial generation of adversarial examples.
-        :type init_size: `int`
         """
         super(BoundaryAttack, self).__init__(estimator=estimator)
 
-        params = {
-            "targeted": targeted,
-            "delta": delta,
-            "epsilon": epsilon,
-            "step_adapt": step_adapt,
-            "max_iter": max_iter,
-            "num_trial": num_trial,
-            "sample_size": sample_size,
-            "init_size": init_size,
-            "batch_size": 1,
-        }
-        self.set_params(**params)
-
-    def generate(self, x, y=None, **kwargs):
+        self.targeted = targeted
+        self.delta = delta
+        self.epsilon = epsilon
+        self.step_adapt = step_adapt
+        self.max_iter = max_iter
+        self.num_trial = num_trial
+        self.sample_size = sample_size
+        self.init_size = init_size
+        self.batch_size = 1
+        self._check_params()
+
+    def generate(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> np.ndarray:
         """
         Generate adversarial samples and return them in an array.
 
         :param x: An array with the original inputs to be attacked.
-        :type x: `np.ndarray`
         :param y: Target values (class labels) one-hot-encoded of shape (nb_samples, nb_classes) or indices of shape
                   (nb_samples,). If `self.targeted` is true, then `y` represents the target labels.
-        :type y: `np.ndarray` or `None`
         :param x_adv_init: Initial array to act as initial adversarial examples. Same shape as `x`.
         :type x_adv_init: `np.ndarray`
         :return: An array holding the adversarial examples.
-        :rtype: `np.ndarray`
         """
         y = check_and_transform_label_format(y, self.estimator.nb_classes, return_one_hot=False)
 
         # Get clip_min and clip_max from the classifier or infer them from data
-        if self.estimator.clip_values is not None:
+        if hasattr(self.estimator, "clip_values") and self.estimator.clip_values is not None:
```

Comment:
clip_values is a property of estimators.
```suggestion
        if self.estimator.clip_values is not None:
```